packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+import { ISequencedDeltaOpMessage } from "./contracts";
+import { ReadBuffer } from "./zipItDataRepresentationReadUtils";
+import { BlobCore, getAndValidateNodeProps, NodeCore, TreeBuilder } from "./zipItDataRepresentationUtils";
+
+export const snapshotMinReadVersion = "1.0";
+
+/**
+ *  Header of the downloaded snapshot.
+ */
+export interface ISnapshotHeader {
+    // This is the minimum version of the reader required to read the wire format of a given snapshot.
+    MinReadVersion: string,
+    // Represents the version with which snapshot is created.
+    CreateVersion: string,
+    // Seq number at which the snapshot is created.
+    SnapshotSequenceNumber: number,
+}
+
+/**
+ * Recreates header section and then validate that.
+ * @param node - tree node to read header section from
+ */
+function readAndValidateHeaderSection(node: NodeCore) {
+    const header =
+        getAndValidateNodeProps<ISnapshotHeader>(node, ["MinReadVersion", "CreateVersion", "SnapshotSequenceNumber"]);
```

Comment:
I do not think this cast is buying us anything - this function should just return Record<> and let callers do appropriate casting.
That said, can we use some TS trickery to get a list of property names out of type (ISnapshotHeader) and not have developer generate them manually as an array? Can you please run it by Tony?

Reply:
Removed the cast and getting the record from the utility function. We can not achieve the functionality of getting names as list from the interface.